### PR TITLE
make use of http.FileServer

### PR DESCRIPTION
**Pull request by hallgren**
Make use of http.FileServer to serve static files.

### CODE REVIEW

**Review by steabert — COMMENTED** [review]
How does http.FileServer handle pre-gzipped files? As far as I can see, it just serves files as they are, without doing any kind of file extension magic?

**Comment by hallgren** [comment]
I added two tests that verifies that the content-type is correct for a html and gz file.

**Comment by steabert** [comment]
When a request for `test.js` comes and the client accepts gzip, then the server should find `test.js.gz` and serve that. As far as I can see, the `http.FileServer` does not do that, it serves the files only if the requested path is a file itself.

File: lib/static_test.go [hunk 1]
```diff
@@ -39,6 +40,66 @@ func TestExistent(t *testing.T) {
...
+
+func TestContentType2(t *testing.T) {
+	middleware := lib.NewStaticMiddleware("../testdir")
+
+	ts := httptest.NewServer(middleware(nil))
+	rsp, err := http.Get(ts.URL + "/test.js.gz")
+	if err != nil {
+		log.Fatal(err)
+	}
+	ce := rsp.Header.Get("Content-Type")
```
Comment (steabert) [inline lib/static_test.go:98]: Here, content-type should be application/javascript, and content-encoding should be gzip (with the server before the changes).

File: lib/static_test.go [hunk 2]
```diff
@@ -39,6 +40,66 @@ func TestExistent(t *testing.T) {
...
+	if rsp.StatusCode != 200 {
+		log.Fatal("expected file to be found")
+	}
+}
+
+func TestContentType2(t *testing.T) {
+	middleware := lib.NewStaticMiddleware("../testdir")
+
+	ts := httptest.NewServer(middleware(nil))
+	rsp, err := http.Get(ts.URL + "/test.js.gz")
```
Comment (steabert) [inline lib/static_test.go:94]: Request should be `/test.js`

**Comment by hallgren** [comment]
I think I misunderstood how this was thought to function.

**Pull request closed without merging**